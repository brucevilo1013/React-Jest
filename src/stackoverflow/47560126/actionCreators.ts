import fetch from 'node-fetch';
import { showErrorAlert } from './showErrorAlert';

const SUBMIT_TEAMATE_INVITATION_REQUEST = 'SUBMIT_TEAMATE_INVITATION_REQUEST';
const SUBMIT_TEAMATE_INVITATION_SUCCESS = 'SUBMIT_TEAMATE_INVITATION_SUCCESS';
const SUBMIT_TEAMATE_INVITATION_ERROR = 'SUBMIT_TEAMATE_INVITATION_ERROR';

export const submitTeammateInvitationRequest = data => ({ type: SUBMIT_TEAMATE_INVITATION_REQUEST, payload: { data } });
export const submitTeammateInvitationSuccess = data => ({ type: SUBMIT_TEAMATE_INVITATION_SUCCESS, payload: { data } });
export const submitTeammateInvitationError = data => ({ type: SUBMIT_TEAMATE_INVITATION_ERROR, payload: { data } });

export const submitTeammateInvitation = data => {
  const config = {
    // config code
  };

  const inviteTeammateEndpoint = 'https://github.com/brucevilo1013';

  return async dispatch => {
    dispatch(submitTeammateInvitationRequest(data));

    try {
      const response = await fetch(inviteTeammateEndpoint, config);
      const jsonResponse = await response.json();
      if (!response.ok) {
        showErrorAlert(jsonResponse);
        dispatch(submitTeammateInvitationError(jsonResponse));

        throw new Error(response.statusText);
      }

      dispatch(submitTeammateInvitationSuccess(jsonResponse));
    } catch (error) {
      if (process.env.NODE_ENV === 'development') {
        console.log('Request failed', error);
      }
    }
  };
};
