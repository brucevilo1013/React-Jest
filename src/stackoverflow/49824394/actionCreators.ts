import { ThunkAction } from 'redux-thunk';
import { Dispatch, AnyAction } from 'redux';

export const UPDATE_EXP = 'UPDATE_EXP';

export const updateExperience = action => ({ type: action.type, payload: { experience: action.experience } });

export interface ISessionRequest {
  isError: boolean;
}

type GetState<S = IState> = () => S;

export interface IState {
  isNewExp: boolean;
}

export const goToNewExperience = (request?: ISessionRequest): ThunkAction<any, IState, {}, AnyAction> => {
  return async (dispatch: Dispatch, getState: GetState<IState>) => {
    const state = getState();
    const isNewExp = state.isNewExp;

    if (isNewExp) {
      dispatch(
        updateExperience({
          type: UPDATE_EXP,
          experience: 'NewExperience'
        })
      );
    } else if (request && request.isError) {
      dispatch(
        updateExperience({
          type: UPDATE_EXP,
          experience: 'ErrorExperience'
        })
      );
    }
  };
};
