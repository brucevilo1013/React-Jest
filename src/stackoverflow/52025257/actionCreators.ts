import followApi from './followApi';

export const handleFavorite = data => {
  return dispatch => {
    return followApi.handleFavorite(data).then(
      payload => {
        dispatch({ type: 'HANDLE_FAVORITE_SUCCESS', payload });
      },
      err => {
        dispatch({ type: 'ERROR', payload: err });
      }
    );
  };
};
