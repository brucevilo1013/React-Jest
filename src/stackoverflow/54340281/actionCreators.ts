const REMOVE_FILTER = 'REMOVE_FILTER';
const SHIPMENTS_PAGINATION = 'SHIPMENTS_PAGINATION';
const TOGGLE_FILTERS_MODAL = 'TOGGLE_FILTERS_MODAL';
const SET_FILTER = 'SET_FILTER';
const SET_CSV_DATA = 'SET_CSV_DATA';

export const removeFilter = (filterKey: string) => ({ type: REMOVE_FILTER, payload: { filterKey } });
export const shipmentsPagination = (pagination: any) => ({ type: SHIPMENTS_PAGINATION, payload: pagination });
export const toggleFiltersModal = () => ({ type: TOGGLE_FILTERS_MODAL });
export const setFilters = (filters: any) => ({ type: SET_FILTER, payload: filters });
export const setCSVDataAction = data => ({ type: SET_CSV_DATA, payload: data });
