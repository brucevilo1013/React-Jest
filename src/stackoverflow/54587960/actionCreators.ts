export const SetFiltersValue = () => ({ type: 'SET_FILTER' });
