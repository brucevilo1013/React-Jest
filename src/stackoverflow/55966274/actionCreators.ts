import { BG_LOAD_START, BG_LOAD_SUCCESS, BG_LOAD_FAILURE } from './types';
import axios from 'axios';

export const bgLoad = (url, pictureNumber = 0) => {
  return dispatch => {
    dispatch(bgLoadStart());

    return axios
      .get(url, {
        headers: {
          Authorization: `Bearer ${process.env.REACT_APP_API_KEY}`
        }
      })
      .then(res => dispatch(bgLoadSuccess(res.data.photos[pictureNumber], res.data.name, res.data.url)))
      .catch(err => dispatch(bgLoadFailure(err)));
  };
};

const bgLoadStart = () => {
  return {
    type: BG_LOAD_START
  };
};

const bgLoadSuccess = (bgImage, name, url) => {
  return {
    type: BG_LOAD_SUCCESS,
    payload: {
      bgImage,
      name,
      url
    }
  };
};

const bgLoadFailure = error => {
  return {
    type: BG_LOAD_FAILURE,
    payload: {
      ...error
    }
  };
};
