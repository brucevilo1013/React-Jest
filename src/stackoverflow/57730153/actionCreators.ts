import axios from 'axios';

export const getOddGroups = () => {
  return dispatch => {
    return axios
      .get('/api/tables/oddgroups')
      .then(results => {
        dispatch({ type: 'GET_ODD_GROUPS', payload: results.data });
      })
      .catch(err => {
        dispatch({ type: 'GET_ERRORS', payload: err.response.message });
      });
  };
};
