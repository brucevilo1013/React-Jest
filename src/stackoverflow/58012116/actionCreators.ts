export const ASSESSMENT = {
  BOOK: {
    ADD: 'ADD',
    GET: 'GET'
  }
};

export const getBook = data => ({ type: ASSESSMENT.BOOK.GET, payload: { data } });

export const someMethod = ({ collections }) => {
  return dispatch => {
    if (collections.length > 0) {
      for (let index = 0; index < collections.length; index++) {
        const collection = collections[index];
        const { libraryUrl, bookUrl, libraryId } = collection;
        const containerId = Math.random().toString();
        dispatch(getBook({ boolURL: bookUrl, libraryURL: libraryUrl, libraryId }));
        dispatch({
          type: ASSESSMENT.BOOK.ADD,
          payload: { boolURL: bookUrl, libraryId }
        });
      }
    }
  };
};
