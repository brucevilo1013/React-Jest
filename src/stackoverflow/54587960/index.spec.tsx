import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import SearchFilter from './';
import { SetFiltersValue } from './actionCreators';

const initialState = { filtersParams: {} };
type State = typeof initialState;
const mockStore = configureStore<State>();

describe('SearchFilter', () => {
  let store;
  beforeEach(() => {
    store = mockStore(initialState);
  });
  it('t1', () => {
    const searchFilter = shallow(<SearchFilter store={store}></SearchFilter>).dive();
    const searchInputWrapper = searchFilter.dive();
    const searchInput = searchInputWrapper.find('input.bp3-input').first();
    expect(searchInputWrapper.state('keyword')).toBe('');

    const mockedEvent = {
      preventDefault: jest.fn(),
      target: { value: 'foo' }
    };
    searchInput.simulate('change', mockedEvent);
    expect(searchInputWrapper.state('keyword')).toBe('foo');
    expect(store.getActions()).toEqual([SetFiltersValue()]);
    expect(searchInputWrapper.html()).toMatchSnapshot();
  });
});
