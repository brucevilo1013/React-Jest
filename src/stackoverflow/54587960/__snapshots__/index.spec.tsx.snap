// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchFilter t1 1`] = `"<div class=\\"search-w bp3-input-group\\"><span class=\\"bp3-icon bp3-icon-search\\"></span><input type=\\"text\\" class=\\"bp3-input\\" value=\\"foo\\" placeholder=\\"Search input\\" dir=\\"auto\\"/></div>"`;
